tests/Test/src/basic/test_basic_logsumexp.birch

```diff
@@ -0,0 +1,149 @@
+
+/*
+ * Test log-sum-exp implementations in `log_sum_exp` and
+ * `resample_reduce`.
+ */
+program test_logsumexp() {
+  // Generate random weights
+  w:Real[100];
+  for n in 1..100 {
+    w[n] <- simulate_gaussian(0.0, 1.0);
+  }
+
+  // Compare with common two-pass algorithm
+  let y <- log_sum_exp_twopass(w);
+  let w2 <- transform(w, \(x:Real) -> { return 2*w; });
+  let ess <- y*y/log_sum_exp_twopass(w2);
+  if !check_ess_logsumexp(w, ess, y) {
+    exit(1);
+  }
+  
+  // Check overflow
```

Comment:
By way of style, use `/* ... */` for whole line comments and no capitals, so `/* check overflow */` here.

Reply:
A few more examples below.

Reply:
```suggestion
  /* check overflow */
```